sycl/include/CL/sycl/atomic.hpp

```diff
@@ -237,8 +237,7 @@ class atomic {
             Ptr);
     cl_int TmpVal = __spirv_AtomicLoad(
         TmpPtr, SpirvScope, detail::getSPIRVMemorySemanticsMask(Order));
-    cl_float ResVal;
-    detail::memcpy(&ResVal, &TmpVal, sizeof(TmpVal));
+    cl_float ResVal = *reinterpret_cast<const cl_float *>(&TmpVal);
```

Comment:
This `reinterpret_cast` is undefined behavior.  We need to use `bit_cast` here instead.